pkg/custom-provider/provider.go

```diff
@@ -147,7 +147,14 @@ func (p *prometheusProvider) buildQuery(ctx context.Context, info provider.Custo
 	// TODO: use an actual context
 	queryResults, err := p.promClient.Query(ctx, pmodel.Now(), query)
 	if err != nil {
-		klog.Errorf("unable to fetch metrics from prometheus: %v", err)
+		if promErr, ok := err.(*prom.Error); ok { // Check if the error is of type *prom.Error
+			klog.Errorf("unable to fetch metrics from prometheus: %s", promErr.Error())
+			mprom.CustomMetricsFailureCounter.WithLabelValues(fmt.Sprintf("%d", promErr.StatusCode)).Inc()
+		} else {
+			// Generic error handling for other types of errors
+			klog.Errorf("unexpected error fetching metrics from prometheus: %s", err.Error())
```

Comment:
nit: replace `prometheus` with `metrics source` given that we use drop in replacements, e.g. `m3` and `pantheon` that provide the same api, but aren't the same metrics backend as `prometheus`

Reply:
ditto with all the other logs